packages/create-mechanic/function-templates/d3-image/index.js

```diff
@@ -0,0 +1,82 @@
+import * as d3 from "d3";
+
+export const handler = ({ inputs, mechanic }) => {
+  const { width, height, text, color1, color2, radiusPercentage } = inputs;
+
+  const center = [width / 2, height / 2];
+  const radius = ((height / 2) * radiusPercentage) / 100;
+  const angle = Math.random() * 360;
+
+  let SVG = d3.create("svg")
+  .attr("width", width).attr("height", height)
+
+  let arc1 = d3.arc().innerRadius(0).outerRadius(radius)
+  .startAngle(0).endAngle(2*Math.PI/2);
+
+  let arc2 = d3.arc().innerRadius(0).outerRadius(radius)
+  .startAngle(2*Math.PI/2).endAngle(4*Math.PI/2);
+
+  SVG.append("path")
+  .attr("transform", `translate(${center[0]},${center[1]}), rotate(${angle})`)
+  .attr("d", arc1).attr("fill", color1)
+
+  SVG.append("path")
+  .attr("transform", `translate(${center[0]},${center[1]}), rotate(${angle})`)
+  .attr("d", arc2).attr("fill", color2)
```

Comment:
I do feel weird having D3 code not formatted in the "D3 style".

I mean like this:
```js
const svg = d3.create("svg")
    .attr("width", width)
    .attr("height", height)
    .attr("viewBox", [0, 0, width, height])
    .attr("style", "max-width: 100%; height: auto; height: intrinsic;");
```

It's not completely necessary really, this is more of an observation. What do you think?

Reply:
@munusshih looks like you changed this! it looks great!